frontend/amundsen_application/static/js/components/Badges/BadgeList/index.tsx

```diff
@@ -69,6 +69,11 @@ export default class BadgeList extends React.Component<BadgeListProps> {
       <span className="badge-list">
         {alphabetizedBadges.map((badge, index) => {
           let badgeConfig;
+          // search case
+          if (typeof badge === 'string' || badge instanceof String) {
+            const badgeName: string = badge.toString();
+            badgeConfig = getBadgeConfig(badgeName);
+          }
```

Comment:
I would refactor this map callback function to separate the two things we are doing: first getting the BadgeConfig and later return the markup.